### PR TITLE
Use magma primitives instead of mantle

### PR DESCRIPTION
**Pull request by rsetaluri**

### CODE REVIEW

**Review by Kuree — APPROVED** [review]
Changes looks fine to me. I assume it requires magma update?

**Comment by rsetaluri** [comment]
@Kuree yes, just requires the latest magma master, which looks like it just happened: https://github.com/StanfordAHA/aha/pull/1190

I'll re-trigger that aha flow test.

File: tests/common/test_transform.py [hunk 1]
```diff
@@ -5,6 +5,18 @@
 import tempfile
 
 
+def _define_binop(op, width):
+    return magma.Bits[width]._declare_binary_op(op)
```
Comment (leonardt) [inline tests/common/test_transform.py:9]: We may want to add an external API for accessing these (I think the problem is they're only available through the operators right now?)

**Review by leonardt — APPROVED** [review]

File: tests/common/test_transform.py (hunk 1 above)
Comment (rsetaluri) [reply tests/common/test_transform.py:9]: Possibly...this code is "bad"...we should just be using the operators here anyway. This is a result of using old gemstone abstractions which should go away.

That being said, we may want a public API for other clients/reasons.

**Pull request merged**